students/Ruohan/lesson04/kata.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+''' trigrams '''
+import string
+import random
+
+file_name = 'sherlock.txt'
+with open (file_name) as f_handle:
+# find the real content we need from the file
+    line_useful = ''
+    bool_flag = False
+    while True:
+        line = f_handle.readline()
+        bool_flag = bool_flag or (line == 'To Sherlock Holmes she is always THE woman. I have seldom heard\n')
+        if bool_flag:
+            line_useful += line
```

Comment:
building up a big string by adding parts on bit by bit is inefficient in Python. 

Better to build it up in a list, and then `.join()` the list.